setup.py

```diff
@@ -54,21 +55,37 @@
 test_reqs = ["pytest", "pytest-cov"]
 
 
-def get_version():
+def get_version(nightly_build=False):
     VERSIONFILE = join("pymc", "__init__.py")
     lines = open(VERSIONFILE).readlines()
     version_regex = r"^__version__ = ['\"]([^'\"]*)['\"]"
     for line in lines:
         mo = re.search(version_regex, line, re.M)
         if mo:
-            return mo.group(1)
+            version = mo.group(1)
+
+            if nightly_build:
+                from datetime import datetime, timezone
+
+                suffix = datetime.now(timezone.utc).strftime(r".dev%Y%m%d")
+                version += suffix
+
+            return version
+
     raise RuntimeError(f"Unable to find version in {VERSIONFILE}.")
 
 
+if "BUILD_PYMC_NIGHTLY" in os.environ:
+    nightly = True
+    DISTNAME += "-nightly"
```

Comment:
if dist name needs to be mutated with environmental variables then there should be a a similar function to `get_version`